src/plugins/discourse/fetch.js

```diff
@@ -270,6 +266,19 @@ export class Fetcher implements Discourse {
       timestampMs: Date.parse(json.created_at),
       authorUsername: json.details.created_by.username,
     };
+
+    // This shouldn't could cause infinite loops when the API is weird.
+    // As requesting pages beyond the last page will produce a 404.
+    // Pagination here is 1-based, and we already had page 1.
+    let page = 2;
```

Comment:
You're right in that the tests haven't changed and have no coverage for pagination. The current test coverage for the fetcher is based on the snapshots from https://sourcecred-test.discourse.group/ and they still pass because none of it has more than 1 page.

To follow the current testing paradigm of snapshots, we need a topic with >20 posts for `topicWithPosts`. As well as >30 topics (not counting category definitions) for `topicsBumpedSince` pagination.

Reply:
Resolved this in #1455